sway/desktop/output.c

```diff
@@ -915,6 +915,33 @@ static struct sway_container *output_get_active_workspace(
 	return workspace;
 }
 
+bool output_has_opaque_lockscreen(struct sway_output *output,
+		struct sway_seat *seat) {
+	if (!seat->exclusive_client) {
+		return false;
+	}
+
+	struct wlr_layer_surface *wlr_layer_surface;
+	wl_list_for_each(wlr_layer_surface, &server.layer_shell->surfaces, link) {
+		if (wlr_layer_surface->output != output->wlr_output) {
+			continue;
+		}
+		struct wlr_surface *wlr_surface = wlr_layer_surface->surface;
+		if (wlr_surface->resource->client != seat->exclusive_client) {
+			continue;
+		}
+		pixman_box32_t output_box = {
```

Comment:
There's no guarantee the layer surface has position 0,0.

Reply:
This is the output box in output-local coords, so it'll always be 0,0. If the surface isn't at 0,0 then the `pixman_region32_contains_rectangle` check will fail.

Reply:
What I mean is: you need to translate `wlr_surface->current.opaque`

Reply:
This is still relevant